### PR TITLE
Add check for invalid syntax is_sle and is_leap

### PR DESCRIPTION
**Pull request by foursixnine**
Often we have the problem a new branch in the code is added and some usage of functions like is_sle and is_leap is not entirely consistent. While this is fixed, we can simply stop from merging PR's that have such invalid syntax

[asciicast](https://asciinema.org/a/MiitxuogQk0NyKByEV1YMFFWJ)

### CODE REVIEW

**Comment by asdil12** [comment]
Ok for now but we should really make `is_leap('15')` work to be consistent with sle.

**Comment by foursixnine** [comment]
> Ok for now but we should really make `is_leap('15')` work to be consistent with sle.

I agree :)

**Comment by foursixnine** [comment]
> Ok for now but we should really make `is_leap('15')` work to be consistent with sle.

@SergioAtSUSE Shall we create a ticket about this?. While I agree... I think there's a bit more that comes into play... somehow.

**Comment by foursixnine** [comment]
Before merging this... I'd like to be sure that we're not going to have surprises... specially when it comes to Staging... Anybody has a second thought?

File: tools/check_invalid_syntax [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
+        echo "${CHECK_LOADTEST_RESULT}"
         echo "Invalid syntax found for loadtest call. '.pm' extensions are not allowed at test distribution level"
-        return $?
+        exit_code=$(($exit_code+1))
+    fi
+    echo "Finished checking loadtests"
+}
+
+check_versions(){
+    CHECK_IS_LEAP_RESULT=$(${GIT_CMD} "is_leap[\(\s](\"|')\d{2}(?!>\.\d)\1")
```
Comment (kravciak) [inline tools/check_invalid_syntax]: Please help me understand:
 - is_leap sub has `\d{2,}\.\d` regex for checking version format
 - it's pushed to version_utils query `/^(?(?!.*\+$)(?<op>[<>=]|[<>]=))(?<qv>$regex)(?<plus>\+)?$/i` as $regex

What is `>` in `(?!>\.\d)` checking for?

Comment (foursixnine) [reply tools/check_invalid_syntax]: @kravciak it's kinda an annoying check tbh xD but I'm rather checking for a negative match.

**Comment by foursixnine** [comment]
> LGTM. The only question, does this check already pass? Or do we need some code changes?

It passes already https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse/jobs/535682022#L665

**Comment by foursixnine** [comment]
> Is it going to work for yaml schedule as well?

I don't see any of the yaml schedule thingies using any of the involved functions, however if any of the files are using any of the is_sle|leap with at least the most obvious invalid syntax... they will be blocked...

**Comment by foursixnine** [comment]
There are corner cases that are not covered, but main obvious problems should be.

**Comment by SergioAtSUSE** [comment]
> > Ok for now but we should really make `is_leap('15')` work to be consistent with sle.
> 
> @SergioAtSUSE Shall we create a ticket about this?. While I agree... I think there's a bit more that comes into play... somehow.

I am ok with creating a ticket. But, such change will need consensus with openSUSE release managers. I remember many people already tried to change it (including me) and had a lot of complaints.

Anyway, I wouldn't consider it high prio. Once your check is on place, we can avoid wrong uses of the current behavior, which is our real priority.

File: tools/check_invalid_syntax (hunk 1 above)
Comment (SergioAtSUSE) [reply tools/check_invalid_syntax]: I think that proves the need of add a bash comment to explain that the intention of the regexp is to find a specific wrong use.

**Pull request merged**